packages/web-ui/src/lab/Button/Button.props.ts

```diff
@@ -0,0 +1,24 @@
+import { ComponentPropsWithoutRef } from 'react';
+import { Responsive } from '../../types';
+
+export type ButtonProps = {
+  /**
+   * Sets the button's visual variant
+   * @default solid
+   */
+  variant?: 'solid' | 'outline' | 'ghost';
+  /**
+   * Sets the button's colour scheme
+   * @default cyan
+   */
+  colorScheme?: 'cyan' | 'red' | 'green' | 'gold' | 'grey';
```

Comment:
Is there anything in place to stop me using `variant="solid"` with `colorScheme="gold"` or `colorScheme="grey"`? If not, should there be?

Without testing this, it should be possible to do something like:

```ts
enum ColorScheme {
  Cyan = "cyan",
  Red = "red",
  Green = "green,
  Gold = "gold",
  Grey = "grey",
}

enum Variants {
  Solid = "solid",
  Outline = "outline",
  Ghost = "ghost",
}

type BaseButtonProps = { ... } // size, asChild, etc

type ButtonProps = BaseButtonProps & (
  | {
      variant?: Exclude<Variants, Variants.Solid>;
      colorScheme?: ColorScheme;
    }
  | {
    variant?: Variants.Solid;
    colorScheme?: Exclude<ColorScheme, ColorScheme.Gold | ColorScheme.Grey>
  }
)
```

Reply:
There was a lot of internal discussion around the existence of these _dead combinations_. 
In short, no, there's nothing stopping you from combining them. However, as usage of the button should be design led, there shouldn't ever be reason to combine them. 

I did start off with something similar to the code you've posted, using TS as a guard, but the error messages were so unhelpful I felt the trade-off wasn't worth it, and could potentially cause more problems.

If there's ever a situation where one of these _dead combinations_ is used, deliberately or accidentally, then the button will be rendered without any colours at all, which I feel is a pretty good visual guard (though am open to other suggestions, ie. should we render the default button 🤔 ). 

I fully accept this situation is not ideal. It exists as we are still in an exploratory stage, we want enough buttons to exist to enable designers and not block devs. In time we will have a deeper understanding of how these buttons are used, and should have more focused requirements, to the extent we may be able to replace all 3 variants of the red colorscheme button with a single `<Button variant="destructive">` etc.

Reply:
I think at this time, I'd rather leave it, and let the component bed in a bit before adding something like this. I'd like to get a better idea of why _"dead prop combinations"_ might occur, and provide a specific solution to that than put this in "just in case".